File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        Map<Character, Character> lookup = new HashMap<>(3);

        lookup.put(')', '(');
        lookup.put('}', '{');
        lookup.put(']', '[');

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (lookup.containsKey(c)) {
                if (!stack.isEmpty() && (lookup.get(c)).equals(stack.peek())) {
                    stack.pop();
                } 
                else return false;
            }

            else stack.push(c);
        }

        return stack.isEmpty();
        
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0] * len(temperatures)
        stack = []


        for i, t in enumerate(temperatures):
            while stack and t > stack[-1][1]:
                i2, t2 = stack.pop()
                res[i2] = i - i2
            stack.append([i, t])

        return res
        